src/app/pages/components/components.ts

```diff
@@ -1,18 +1,64 @@
-import { Component} from '@angular/core';
+import {Component} from '@angular/core';
+
+export class ComponentGroup {
+  name: string;
+  list: ComponentItem[];
+
+  constructor(name: string, list: ComponentItem[]) {
+    this.name = name;
+    this.list = list;
+  }
+}
+export class ComponentItem {
+  name: string;
+  src: string;
+  link: string;
+
+  constructor(name: string, src?: string, link?: string) {
+    this.name = name;
+    this.src = src ? src : name;
+    this.link = link ? link : this.src;
+  }
+
+  get routerLink() {
+    return '../component/' + this.link;
+  }
+
+  get imageSrc() {
+    return `../../../assets/img/components/${this.src}.svg`;
+  }
+}
 
 @Component({
   selector: 'app-components',
   templateUrl: 'components.html',
-  styleUrls: ['components.scss']
+  styleUrls: ['components.scss'],
 })
 export class ComponentsList {
-  componentItems = [
-    {name: 'Buttons', src: 'button'},
-    {name: 'Cards', src: 'card'},
-    {name: 'Chips', src: 'chip'},
-    {name: 'Grid lists', src: 'grid'},
-    {name: 'Menu', src: 'menu'},
-    {name: 'Tooltip', src: 'tooltip'},
-    {name: 'Progress', src: 'progress'},
+  componentGroups: ComponentGroup[] = [
```

Comment:
I think this would be a bit more readable if we defined the data like this:
```ts
componentGroups = groupComponents([
  {name: 'Buttons',  src: 'button',   group: 'Form controls'},
  {name: 'Checkbox', src: 'checkbox', group: 'Form controls'},
  ...
]);
```
Where `groupComponents` creates the data structures. Thoughts?